src/server/mod.rs

```diff
@@ -1664,6 +1666,189 @@ impl LanguageServer for LspServer {
+                                    "Failed to update composition."
+                                        .to_string(),
+                                )
+                                .into());
+                            }
+                            composition.to_string()
+                        } else {
+                            return Err(LspError::InternalError(
+                                "Missing composition".to_string(),
+                            )
+                            .into());
+                        }
+                    }
+                    Err(_) => {
+                        return Err(LspError::LockNotAcquired.into())
+                    }
+                };
+
+                let last_pos =
+                    line_col::LineColLookup::new(&new_text)
+                        .get(new_text.len());
+
+                let edit = lsp::WorkspaceEdit {
+                    changes: Some(HashMap::from([(
+                        command_params.text_document.uri.clone(),
+                        vec![lsp::TextEdit {
+                            new_text: new_text.clone(),
+                            range: lsp::Range {
+                                start: lsp::Position::default(),
+                                end: lsp::Position {
+                                    line: last_pos.0 as u32,
+                                    character: last_pos.1 as u32,
+                                },
+                            },
+                        }],
+                    )])),
+                    document_changes: None,
+                    change_annotations: None,
+                };
+
+                if let Some(client) = self.get_client() {
+                    match client.apply_edit(edit, None).await {
+                        Ok(response) => {
+                            if response.applied {
+                                self.store.put(
+                                    &command_params.text_document.uri,
+                                    &new_text,
+                                );
+                            }
+                        }
+                        Err(err) => {
+                            return Err(LspError::InternalError(
+                                format!("{:?}", err),
+                            )
+                            .into())
+                        }
+                    };
+                };
+
+                Ok(None)
+            }
+            Ok(LspServerCommand::RemoveMeasurementFilter) => todo!(),
+            Ok(LspServerCommand::AddFieldFilter) => todo!(),
+            Ok(LspServerCommand::RemoveFieldFilter) => todo!(),
+            Ok(LspServerCommand::AddTagFilter) => todo!(),
+            Ok(LspServerCommand::RemoveTagFilter) => todo!(),
+            Ok(LspServerCommand::AddTagValueFilter) => todo!(),
+            Ok(LspServerCommand::RemoveTagValueFilter) => todo!(),
```

Comment:
I don't think we should make a habit out of committing these. This one time it's probably okay, but in the future, we should not do it.